devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy

```diff
@@ -12,406 +12,259 @@
  *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
  *  for the specific language governing permissions and limitations under the License.
  *
- -------10/20/2015 Updates--------
- -Fix/add battery reporting interval to update
- -remove polling and/or refresh
- 
- -------5/2017 Updates--------
- -Add fingerprints for SLP
- -add device health, check every 60mins + 2mins
- 
- -------3/2019 Updates--------
- -Add fingerprints for SLP3
- -change device health from 62mins to 3 hours
+
+ -------6/2021 Updates--------
+ - Update for 2021 Samsung SmartThings App
+
  */
- 
+
+import groovy.json.JsonOutput
+import physicalgraph.zigbee.zcl.DataType
+
+//dth version
+def getVERSION() {"v1.0 6-2021"}
+def getDEBUG() {true}
+def getHC_INTERVAL_MINS() {60}
+
 metadata {
-	definition (name: "Spruce Sensor", namespace: "plaidsystems", author: "Plaid Systems") {
-		
-		capability "Configuration"
+	definition (name: "Spruce Sensor", namespace: "plaidsystems", author: "Plaid Systems", mnmn: "SmartThingsCommunity",
+		mcdSync: true, vid: "4cff4731-67ce-310b-ada0-4d8e169a6df0") {
+
+		capability "Sensor"
+		capability "Temperature Measurement"
+		capability "Relative Humidity Measurement"
 		capability "Battery"
-        capability "Relative Humidity Measurement"
-        capability "Temperature Measurement"
-        capability "Sensor"
-        capability "Health Check"
-        //capability "Polling"
-		
-        attribute "maxHum", "string"
-        attribute "minHum", "string"        
-        
-        
-        command "resetHumidity"
-        command "refresh"
-        
-        fingerprint profileId: "0104", inClusters: "0000,0001,0003,0402,0405", outClusters: "0003, 0019", manufacturer: "PLAID SYSTEMS", model: "PS-SPRZMS-01", deviceJoinName: "Spruce Irrigation" //Spruce Sensor
-        fingerprint profileId: "0104", inClusters: "0000,0001,0003,0402,0405", outClusters: "0003, 0019", manufacturer: "PLAID SYSTEMS", model: "PS-SPRZMS-SLP1", deviceJoinName: "Spruce Irrigation" //Spruce Sensor
-        fingerprint profileId: "0104", inClusters: "0000,0001,0003,0402,0405", outClusters: "0003, 0019", manufacturer: "PLAID SYSTEMS", model: "PS-SPRZMS-SLP3", deviceJoinName: "Spruce Irrigation" //Spruce Sensor
+		capability "Health Check"
+		capability "Configuration"
+		capability "Refresh"
+
+		attribute "reportingInterval", "NUMBER"
+
+		//new release
+		fingerprint manufacturer: "PLAID SYSTEMS", model: "PS-SPRZMS-01", zigbeeNodeType: "SLEEPY_END_DEVICE", deviceJoinName: "Spruce Irrigation Sensor"
+		fingerprint manufacturer: "PLAID SYSTEMS", model: "PS-SPRZMS-SLP1", zigbeeNodeType: "SLEEPY_END_DEVICE", deviceJoinName: "Spruce Irrigation Sensor"
+		fingerprint manufacturer: "PLAID SYSTEMS", model: "PS-SPRZMS-SLP3", zigbeeNodeType: "SLEEPY_END_DEVICE", deviceJoinName: "Spruce Irrigation Sensor"
```

Comment:
I'm afraid you'll have to stick to current `deviceJoinName`s

Reply:
changed back to "Spruce Irrigation"